copilot-instructions.md

```diff
@@ -0,0 +1,232 @@
+- Optional: `api-tools` for API diffs: `dotnet tool install -g api-tools`
+
+### Common Build Commands
+```bash
+# Full build of all packages
+dotnet cake
+
+# Clean build for CI
+dotnet cake -t=ci
+
+# Build specific target
+dotnet cake --target=libs
+dotnet cake --target=nuget
+dotnet cake --target=packages
+
+# Update config to latest Maven versions
+dotnet cake --target=update-config
+
+# Bump NuGet package versions (4th component)
+dotnet cake --target=bump-config
+
+# Generate API differences
+dotnet cake nuget-diff.cake
+
+# Run utilities (governance, mappings, etc.)
+dotnet cake utilities.cake
+```
+
+## Configuration System
+
+### config.json Structure
+Each artifact entry contains:
+```json
+{
+  "groupId": "androidx.activity",
+  "artifactId": "activity", 
+  "version": "1.10.1",
+  "nugetVersion": "1.10.1.2",
+  "nugetId": "Xamarin.AndroidX.Activity",
+  "dependencyOnly": false  // true for transitive deps only
+}
+```
+
+### Version Conventions
+- **Major.Minor.Patch**: Mirrors the Maven artifact version
+- **Revision (4th component)**: Used for NuGet-only updates without Maven changes
+- **Pre-release suffixes**: Supported (e.g., "1.0.0.1-alpha05")
+
+## Development Workflow
+
+### Adding New Bindings
+1. Add artifact entry to `config.json`
+2. Run `dotnet cake --target=binderate` to generate projects
+3. Add any necessary customizations in `source/{groupId}/{artifactId}/`
+4. Build and test: `dotnet cake --target=libs`
+
+### Updating Existing Bindings
+1. Use `dotnet cake --target=update-config` for automatic updates
+2. Or manually edit versions in `config.json`
+3. Run `dotnet cake --target=binderate-diff` to see changes
+4. Build and validate
+
+### Version Bumping
+- Use `dotnet cake --target=bump-config` to increment revision numbers
+- This updates only non-dependency packages (where `dependencyOnly != true`)
+
+### Release Process
+1. Tag commit: `git tag YYYYMMDD-weekly-stable-updates-YYYYMMDD`
+2. Push tag: `git push upstream <tag>`
+3. Use AndroidX Push NuGet.org pipeline in Azure DevOps
+
+## Binding Customizations
+
+### Per-Artifact Customizations
+Located in `source/{groupId}/{artifactId}/`:
+- **`Additions/`**: Additional C# code to add to generated bindings
+- **`Transforms/`**: XML transforms to modify generated API
+- **`Metadata.xml`**: Binding metadata and parameter names
+- **`merge.targets`**: Custom MSBuild targets to include
+
+### Common Metadata Patterns
+```xml
+<!-- Remove problematic APIs -->
+<remove-node path="/api/package[@name='com.example']/class[@name='ProblematicClass']" />
+
+<!-- Fix parameter names -->
+<attr path="/api/package[@name='com.example']/class[@name='Example']/method[@name='example']/parameter[@name='p0']" name="name">properName</attr>
+
+<!-- Add managed wrapper -->
+<add-node path="/api/package[@name='com.example']">
+  <class abstract="false" deprecated="not deprecated" final="false" name="ManagedWrapper" static="false" visibility="public">
+  </class>
+</add-node>
+```
```

Comment:
Add this link in here, it's probably the most important doc:
* https://github.com/dotnet/java-interop/wiki/Troubleshooting-Android-Bindings-Issues